Engine/Component/ComponentScript.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ComponentScript.h"
+#include "Main/Application.h"
+#include "Module/ModuleRender.h"
+#include "Main/GameObject.h"
+#include "Module/ModuleScriptManager.h"
+#include "Script/Script.h"
+#include "imgui.h"
+
+ComponentScript::ComponentScript() : Component(nullptr, ComponentType::SCRIPT)
+{
+
+}
+ComponentScript::ComponentScript(GameObject * owner, std::string script_name) : Component(owner, ComponentType::SCRIPT)
+{
+	script = App->scripts->CreateResourceScript(script_name, owner);
+	App->scripts->scripts.push_back(this);
+}
+
+ComponentScript::~ComponentScript()
+{
+}
+
+void ComponentScript::LoadName(std::string script_name) 
+{
+	this->name = script_name;
+	script = App->scripts->CreateResourceScript(script_name, owner);
+}
+
+void ComponentScript::Update()
+{
+	if (script) 
+	{
+		script->Update();
+	}
+}
+
+void ComponentScript::Delete()
+{
+	App->scripts->RemoveComponentScript(this);
+}
+
+void ComponentScript::ShowComponentWindow()
+{
+	if (script)
+	{
+		script->OnInspector(ImGui::GetCurrentContext());
+	}
+	else {
+		APP_LOG_ERROR("wtf i'm doing");
+	}
+}
+
+void ComponentScript::Save(Config & config) const
```

Comment:
Another fancy stylish thing, the & should be inmediatly after the type.
`void ComponentScript::Save(Config& config) const`

Reply:
Done!